pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -705,14 +717,34 @@ func (p *Pod) cleanupExcessPods(ctx context.Context, c client.Client, totalCount
 	if extraPodsCount <= 0 {
 		return nil
 	}
+	// Do not clean up more pods until observing previous operations
+	if !p.satisfiedExcessPods {
+		return errPendingOps
+	}
 
 	// Sort active pods by creation timestamp
 	sort.Slice(activePods, func(i, j int) bool {
-		return activePods[i].ObjectMeta.CreationTimestamp.Before(&activePods[j].ObjectMeta.CreationTimestamp)
+		pi := &activePods[i]
+		pj := &activePods[j]
+		iFin := slices.Contains(pi.Finalizers, PodFinalizer)
+		jFin := slices.Contains(pj.Finalizers, PodFinalizer)
+		// Keep pods that have a finalizer
```

Comment:
Does it really mean `keep` or, just prioritize removal of pods without finalizer?

Reply:
updated comment

Reply:
sgtm